apps/sms/views/conversation/js/conversation.js

```diff
@@ -1840,6 +1860,46 @@ var ConversationView = {
       }) : '';
   },
 
+  /**
+   * @param  {Object} A SMS/MMS message
+   * @return {String} The late arrival notice string
+   */
+  getLateArrivalNotice: function(message) {
+    var opts = this.LATE_NOTICE_OPTS;
+    var sent = message.sentTimestamp;
+    var received = message.timestamp;
+    var lateness = received - sent;
+    var period = MONTH;
+    var noticeL10nIds = {
+      [MONTH]: 'late-arrival-notice-in-months',
+      [DAY]: 'late-arrival-notice-in-days',
+      [HOUR]: 'late-arrival-notice-in-hours',
+      [MINUTE]: 'late-arrival-notice-in-minutes'
+    };
+
+    if (!sent || !received || lateness < opts.OFF_NOTICE_LOWER_LIMIT) {
+      return '';
+    }
+
+    if (lateness < opts.IN_HOURS_LIMIT)       { period = MINUTE; } 
+    else if (lateness < opts.IN_DAYS_LIMIT)   { period = HOUR; } 
+    else if (lateness < opts.IN_MONTHS_LIMIT) { period = DAY; }
+    
+    var noticeNode = this.tmpl.lateArrivalNotice.prepare({
+        noticeL10nId: noticeL10nIds[period]
+    }).toDocumentFragment();
+
+    navigator.mozL10n.setAttributes(
+      noticeNode.querySelector('.late-arrival-notice'), 
+      noticeL10nIds[period], 
+      {
+        'lateness': (lateness - lateness % period) / period
+      }
+    );
+
+    return noticeNode.firstChild.outerHTML;
```

Comment:
@julienw I don't like that either :) I was wondering if the mozL10n had an API that given an ID and some data would translate and return a simple string. These string could be inserted later in the template mechanism. Maybe it is [`L10n.get`](https://developer.mozilla.org/en-US/docs/Web/API/L10n) but there is no documentation for the method signature. Or, as you said, I can return the fragment and insert it manually.
What do you think ?